restricted_site_access.php

```diff
@@ -122,6 +122,32 @@ public static function add_actions() {
 
 		add_filter( 'pre_option_blog_public', array( __CLASS__, 'pre_option_blog_public' ), 10, 1 );
 		add_filter( 'pre_site_option_blog_public', array( __CLASS__, 'pre_option_blog_public' ), 10, 1 );
+		add_filter( 'application_password_is_api_request', array( __CLASS__, 'is_api_request' ) );
+	}
+
+	/**
+	 * Add filter to override application_password_is_api_request filter. This has to be done since rest API isn't been loaded until we call restrict_access.
```

Comment:
```suggestion
	 * Determine if this is a REST request.
	 * 
	 * Determine whether this is a REST API request based on the URL. As RSA redirects prior
	 * to the `init` hook running, RSA needs to replace the API check in wp_authenticate_application_password().
	 *
	 * @since x.x.x
```

Nitpick because I can't help myself.